libgstc/pygstc/gstc.py

```diff
@@ -0,0 +1,840 @@
+            elif type_list[i] == bool:
+                if parameter:
+                    parameter_string_list += ['true']
+                else:
+                    parameter_string_list += ['false']
+            else:
+                parameter_string_list += [str(parameter)]
+        return parameter_string_list
+
+    def _send_cmd_line(self, cmd_line):
+        """
+        Send a command using an abstract IPC and wait for the response.
+
+        Parameters
+        ----------
+        cmd_line : string list
+            Command to be send
+
+        Raises
+        ------
+        GstdError
+            Error is triggered when Gstd IPC fails
+        GstcError
+            Error is triggered when the Gstd python client fails internally
+
+        Returns
+        -------
+        result : dictionary
+            Response from the IPC
+        """
+
+        cmd = cmd_line[0]
+        try:
+            jresult = self._ipc.send(cmd_line, timeout=self._timeout)
+            result = json.loads(jresult)
+        except Exception as exception:
+            self._logger.error('%s error: %s' % (cmd,
+                                                 type(exception).__name__))
+            traceback.print_exc()
+            raise GstcError(type(exception).__name__)
+        if result['code'] != 0:
+            self._logger.error('%s error: %s' % (cmd,
+                                                 result['description']))
+            raise GstdError(result['description'])
+        return result
+
+    def ping_gstd(self):
+        """
+        Test if GSTD responds in the configured address and port
+
+        Returns
+        -------
+        result : bool
+            Whether or not GSTD responds
+        """
+        self._logger.info('Sending ping to GStreamer Daemon')
+        try:
+            jresult = self._ipc.send(['list_pipelines'], timeout=1)
+            result = json.loads(jresult)
+            if 'description' in result:
+                if result['description'] != "Success":
+                    raise Exception
+            else:
+                raise Exception
+        except Exception as exception:
+            self._logger.error('GStreamer Daemon failed to respond')
+            return False
+        return True
+
+    def bus_filter(self, pipe_name, filter):
+        """
+        Select the types of message to be read from the bus. Separate
+        with a '+', i.e.: eos+warning+error.
+
+        Parameters
+        ----------
+        pipe_name: string
+            The name of the pipeline
+        filter: string
+            Filter to be applied to the bus. '+' reparated strings
+        """
+
+        self._logger.info('Setting bus read filter of pipeline %s to %s'
+                          % (pipe_name, filter))
+        parameters = self._check_parameters([pipe_name, filter], [str, str])
+        self._send_cmd_line(['bus_filter'] + parameters)
+
+    def bus_read(self, pipe_name):
```

Comment:
We are leaving the responsibility of async calls and multithreading to the user, to keep the client lighter and consistent. I think that it is confusing that some call would spawn a thread and others not. Also, we would need to add an argument to keep the option to call this as a blocking function in case that is what the user wants, and that goes against having a cohesive interface between this client and `gstd-client`.

Reply:
I disagree, I'd have both. Waiting on the bus on a separate thread is a very common pattern. Please open a feature for next release.